### PR TITLE
CCC: Switch Enum values for spec.stateIntoSpec to upper camel case

### PR DESCRIPTION
**Pull request by justinsb**
This follows the KRM conventions.

### CODE REVIEW

File: pkg/k8s/defaulter.go [hunk 1]
```diff
@@ -44,48 +43,37 @@ func NewStateIntoSpecDefaulter(client client.Client) Defaulter {
...
 			return false, fmt.Errorf("error getting ConfigConnectorContext object %v/%v: %w", cccNamespacedName.Namespace, cccNamespacedName.Name, err)
 		}
-	} else {
-		stateIntoSpecOverridePtr = ccc.Spec.StateIntoSpec
 	}
-	defaultValue, err := v.getDefaultValue(stateIntoSpecOverridePtr, StateIntoSpecDefaultValueV1Beta1)
-	if err != nil {
-		return false, fmt.Errorf("error getting the default value: %w", err)
+
+	if ccc != nil && ccc.Spec.StateIntoSpec != nil {
```
Comment (maqiuyujoyce) [inline pkg/k8s/defaulter.go:61]: Merging in the defaulting logic can avoid additional checks, makes sense! Thank you for the refactoring.

**Review by maqiuyujoyce — COMMENTED** [review]
LGTM! Have you verified the change locally (via the integration test etc)?

**Comment by justinsb** [comment]
>  locally (via the integration test etc)?

Our tests run on every PR!

**Comment by justinsb** [comment]
I'm going to try to find a stepwise test PR into which I can add a test for this; it is weird that I didn't have to update a test case from lower case to upper case now that you point it out @maqiuyujoyce  - thanks!

**Comment by justinsb** [comment]
Added some tests CC @maqiuyujoyce

**Comment by maqiuyujoyce** [comment]
Hi @justinsb , thank you for adding the tests. As thoroughly reviewing the test will take some time, and we want to unblock 1.114 release asap, could you split the test changes out into a separate PR and I can approve this PR quickly? Interface changes LGTM though I want to make sure the test changes are also probably reviewed.

**Comment by justinsb** [comment]
@maqiuyujoyce sure, I removed the test commits and will add to another PR

**Comment by maqiuyujoyce** [comment]
/lgtm
/approve

> Edit: So I'm confused, I didn't find any tests for this codepath. I think we could build one using a step test (because that exports the objects).

Right, there is not yet any test step for it. I plan to add one into the dynamic test flow.

**Pull request merged**